src/Makefile.test.include

```diff
@@ -416,6 +416,31 @@ if EMBEDDED_UNIVALUE
 	$(AM_V_at)$(MAKE) $(AM_MAKEFLAGS) -C univalue check
 endif
 
+if ENABLE_BENCH
+if ENABLE_QT_TESTS
+check-valgrind: $(TEST_BINARY) $(BENCH_BINARY) test_bitcoin_qt
+else
+check-valgrind: $(TEST_BINARY) $(BENCH_BINARY)
+endif
+else
+if ENABLE_QT_TESTS
+check-valgrind: $(TEST_BINARY) test_bitcoin_qt
+else
+check-valgrind: $(TEST_BINARY)
+endif
+endif
+	@echo "Using the valgrind memory error detector: expect an ~50x slowdown, valgrind 3.14 or later required"
+	@echo "Running $(TEST_BINARY) under valgrind -- this will take a quite a while ..."
+	valgrind --suppressions=$(top_builddir)/contrib/valgrind.supp --gen-suppressions=all --exit-on-first-error=yes --error-exitcode=1 --quiet $(TEST_BINARY)
```

Comment:
I'd rather keep running the tests in parallel, even more so that valgrind causes a slowdown. See https://github.com/bitcoin/bitcoin/pull/17490/files#diff-3b1d470e34d4107835b2c4827cde6dc6R40

Reply:
I'm not sure how to do it in parallel in a way that a.) prints generates suppressions to the user, and b.) does not potentially introduce false positives due to resource exhaustion. Perhaps introducing parallelism can be tackled in a follow up PR?

Reply:
a) The suppression should be printed when the test case fails, no?

b) I think we can't prevent OOM in general due to users setting too high `-j` in `make check`. Do you have any hints when this could lead to false positives instead of a killed process or frozen machine?

Reply:
Also, note that the same OOM "protection" does not exist for the functional tests when run with `--valgrind` or so

Reply:
I think these things would be nice to have but I'm afraid I won't have time to add that to this PR any time soon. I suggest we go with this simple version to start with and then someone can add parallellism in a follow-up PR. Makes sense? :)